shell/spy.go

```diff
@@ -2,18 +2,37 @@ package shell
 
 import (
 	"bytes"
-	"log"
+	"fmt"
+	"io"
 	"os/exec"
+
+	"bitbucket.org/creachadair/shell"
 )
 
-// Execute accepts a command argument and returns the result of the executed command
-func Execute(script string) string {
-	cmd := exec.Command(script)
+// ParseCommand accepts a string and returns the command name and args
+func ParseCommand(cmd string) (string, []string) {
+	stringResult, _ := shell.Split(cmd)
+
+	if len(stringResult) <= 1 {
+		return cmd, []string{}
+	}
+	return stringResult[0], stringResult[1:]
+}
+
+// Execute accepts a command argument and writes the result to a writer
+func Execute(w io.Writer, commandLine string) error {
 	var out bytes.Buffer
-	cmd.Stdout = &out
+	name, args := ParseCommand(commandLine)
+	cmd := exec.Command(name, args...)
+
+	multiW := io.MultiWriter(w, &out)
+	cmd.Stdout = multiW
+
 	err := cmd.Run()
 	if err != nil {
-		log.Fatal(err)
+		return err
 	}
-	return out.String()
+
+	fmt.Println(out.String())
```

Comment:
And we shouldn't be printing to the terminal here—if `w` is a multiwriter connected to the terminal, the command's output will have already gone to it, so nothing else is needed.